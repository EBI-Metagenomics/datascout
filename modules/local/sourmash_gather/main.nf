#!/usr/bin/env nextflow
process SOURMASH_GATHER {
    container 'quay.io/biocontainers/sourmash:4.8.14--hdfd78af_0'

    publishDir "${params.output}", mode: "copy"
    debug true
    label "process_high"

    tag "${meta}"

    errorStrategy  'retry'
    maxRetries 2

    input:
      tuple val(meta), path(genome_sig), path(list_of_read_sigs)

    output:
      tuple val(meta), path("*.csv"), emit: gather_csv
      path("versions.yml"), emit: versions


    script:
    """
    sourmash gather ${genome_sig} ${list_of_read_sigs.join(' ')} -o ${meta.id}_${meta.ena_tax}_sourmash_gather.csv
    if [[ ! -f "${meta.id}_${meta.ena_tax}_sourmash_gather.csv" ]]; then
        echo "${meta.id}_${meta.ena_tax} had no matching transcriptomic data. No transcriptomic fastq files will be returned";
        touch "${meta.id}_${meta.ena_tax}_empty.csv";
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        Sourmash: \$( sourmash --version 2>&1 | cut -d' ' -f2 )
    END_VERSIONS
    """
}

