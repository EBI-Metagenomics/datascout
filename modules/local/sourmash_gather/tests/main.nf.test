nextflow_process {

    name "Test Process SOURMASH_GATHER"
    script "modules/local/sourmash_gather/main.nf"
    process "SOURMASH_GATHER"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = tuple(
                [id:"GCA_003719455.1", ena_tax:"default"], 
                "$projectDir/tests/reference_databases/GCA_003719455.1.sig", 
                ["$projectDir/tests/reference_databases/SRR32368313.sig"]
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out.gather_csv,
                process.out.versions
            ).match()
        }

    }

}
