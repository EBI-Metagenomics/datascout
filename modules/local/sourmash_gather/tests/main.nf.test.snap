{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "id": "GCA_003719455.1",
                        "ena_tax": "default"
                    },
                    "GCA_003719455.1_default_sourmash_gather.csv:md5,651c78167cfb4f462b6d3c3d2a24940f"
                ]
            ],
            [
                "versions.yml:md5,2d7ad663426fda66ce69eb2b6778478d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-09-10T19:48:09.36207"
    }
}